funcx_endpoint/funcx_endpoint/executors/high_throughput/executor.py

```diff
@@ -776,6 +785,32 @@ def shutdown(self, hub=True, targets='all', block=False):
         logger.info("Finished HighThroughputExecutor shutdown attempt")
         return True
 
+    def _cancel(self, future):
+        logger.debug("Sending cancel of task_id:{future.task_id} to interchange")
+        response = self.command_client.run(TaskCancel(future.task_id))
+        logger.debug("Sent TaskCancel to interchange")
+        return response
+
+
+CANCELLED = 'CANCELLED'
+CANCELLED_AND_NOTIFIED = 'CANCELLED_AND_NOTIFIED'
+FINISHED = 'FINISHED'
+
+
+class FuncXFuture(Future):
+
+    def __init__(self, executor):
+
+        super().__init__()
+        self.executor = executor
+
+    def cancel(self):
+
+        self.executor._cancel(self)
+        super().cancel()
+
+        return True
```

Comment:
This is a tough one. Cancelling is full of potential race-conditions. What ought to happen if the task was finished and results were on the way back when the user invoked `future.cancel()`?
Trying to make this perfect will burn a lot of engineering for marginal improvement here. We can be clear in our docs that cancel is best effort with the goal of avoiding wasted compute. From the user interface perspective, I think we should return False if the task is already complete, but otherwise returning True is the right thing to do.